cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -4216,31 +4217,62 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             // Lower to VM calls when there's no access to SSE4.1.
             let ty = ty.unwrap();
             if !ty.is_vector() {
-                let libcall = match (op, ty) {
-                    (Opcode::Ceil, types::F32) => LibCall::CeilF32,
-                    (Opcode::Ceil, types::F64) => LibCall::CeilF64,
-                    (Opcode::Floor, types::F32) => LibCall::FloorF32,
-                    (Opcode::Floor, types::F64) => LibCall::FloorF64,
-                    (Opcode::Nearest, types::F32) => LibCall::NearestF32,
-                    (Opcode::Nearest, types::F64) => LibCall::NearestF64,
-                    (Opcode::Trunc, types::F32) => LibCall::TruncF32,
-                    (Opcode::Trunc, types::F64) => LibCall::TruncF64,
-                    _ => panic!(
-                        "unexpected type/opcode {:?}/{:?} in Ceil/Floor/Nearest/Trunc",
-                        ty, op
-                    ),
-                };
-                emit_vm_call(ctx, flags, triple, libcall, insn, inputs, outputs)?;
+                if isa_flags.has_sse41() {
+                    let (op, mode) = match (op, ty) {
```

Comment:
Hi @bnjbvr .. The match here is already using two parameters, an opcode and type so I am missing what you mean here. Do you mean I could include the vector types in the same match statement so that I am checking for both types::F32 and types::F32X4, etc. That's how I will interpret it since I think that would be cleaner logic that would avoid duplication at the source level.

Reply:
@bnjbvr latest push maybe does what is suggested here but let me know.

Reply:
Ah, that's nice! At least it'll make the error for vector opcodes very explicit by panicking.

Sorry for being unclear. I was actually suggesting to break the `match (op, ty) { ... }` into two: first, a `match op` that'll give us the `RoundImm`, then a `match ty` that will give us the `SseOpcode`. We can do this to make the code more readable, since determining the SSE opcode is an independent task from determining the rounding immediate. Does it make sense?

Reply:
@bnjbvr .. Hi I got you .. sorry about misunderstanding. I think I've made that improvement now this latest update. May be good to go now.